### ISSUE TITLE
[BUG] CircleCI orb setup step fails ever since 7.55 was released

### ISSUE THREAD

**Issue by saada** [DataDog/datadog-agent#27523]
**Agent Environment**

7.55

**Describe what happened:**
Using the orb, it auto-installs version 7.55 and there is no way to pin the MINOR versions; only MAJOR version 7.
The orb started throwing this error

```
datadog-agent: unrecognized service
```

We noticed in the release notes of https://github.com/DataDog/datadog-agent/releases/tag/7.55.0 this note that could be related

```
Improve concurrency handling for Agent start/simultaneous restart. Agent starting and restarting concurrent execution may happen during the Agent's installation or upgrade.
```

**Describe what you expected:**
Using the previous 7.54.1 it showed

```
 * Starting Datadog Agent datadog-agent
   ...done.
 * Starting Datadog Process Agent datadog-agent-process
   ...done.
 * Starting Datadog Trace Agent (APM) datadog-agent-trace
   ...done.
```

**Steps to reproduce the issue:**
```
  install-datadog:
    steps:
      - datadog-agent/setup:
          agent_major_version: '7'
          api_key: DATADOG_API_KEY
          site: datadoghq.com
```

We should either have the ability to pin the MINOR versions or fix the installation process in the orb.

<details>
  <summary>View installation logs</summary>

  ### Full installation log
  ```js
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 33349  100 33349    0     0   320k      0 --:--:-- --:--:-- --:--:--  322k

 install_script.sh is deprecated. Please use one of
 
 * https://s3.amazonaws.com/dd-agent/scripts/install_script_agent6.sh to install Agent 6
 * https://s3.amazonaws.com/dd-agent/scripts/install_script_agent7.sh to install Agent 7


* Datadog Agent install script v1.13.0


* Installing apt-transport-https, curl and gnupg

Get:1 https://download.docker.com/linux/ubuntu jammy InRelease [48.8 kB]
Get:2 https://dl.google.com/linux/chrome/deb stable InRelease [1,825 B]
Get:3 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]
Get:4 http://archive.ubuntu.com/ubuntu jammy InRelease [270 kB]
Get:5 https://download.docker.com/linux/ubuntu jammy/stable amd64 Packages [42.1 kB]
Get:6 https://dl.google.com/linux/chrome/deb stable/main amd64 Packages [1,091 B]
Get:7 https://packagecloud.io/github/git-lfs/ubuntu jammy InRelease [28.0 kB]
Get:8 https://ppa.launchpadcontent.net/git-core/ppa/ubuntu jammy InRelease [24.6 kB]
Get:9 https://ppa.launchpadcontent.net/mozillateam/ppa/ubuntu jammy InRelease [24.6 kB]
Get:10 https://packagecloud.io/github/git-lfs/ubuntu jammy/main amd64 Packages [1,842 B]
Get:11 http://security.ubuntu.com/ubuntu jammy-security/universe amd64 Packages [1,127 kB]
Get:12 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]
Get:13 https://ppa.launchpadcontent.net/git-core/ppa/ubuntu jammy/main amd64 Packages [2,972 B]
Get:14 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]
Get:15 http://archive.ubuntu.com/ubuntu jammy/universe amd64 Packages [17.5 MB]
Get:16 https://ppa.launchpadcontent.net/mozillateam/ppa/ubuntu jammy/main amd64 Packages [43.6 kB]
Get:17 http://security.ubuntu.com/ubuntu jammy-security/multiverse amd64 Packages [44.7 kB]
Get:18 http://security.ubuntu.com/ubuntu jammy-security/restricted amd64 Packages [2,601 kB]
Get:19 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [1,998 kB]
Get:20 http://archive.ubuntu.com/ubuntu jammy/multiverse amd64 Packages [266 kB]
Get:21 http://archive.ubuntu.com/ubuntu jammy/restricted amd64 Packages [164 kB]
Get:22 http://archive.ubuntu.com/ubuntu jammy/main amd64 Packages [1,792 kB]
Get:23 http://archive.ubuntu.com/ubuntu jammy-updates/multiverse amd64 Packages [51.8 kB]
Get:24 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 Packages [1,410 kB]
Get:25 http://archive.ubuntu.com/ubuntu jammy-updates/restricted amd64 Packages [2,674 kB]
Get:26 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [2,263 kB]
Get:27 http://archive.ubuntu.com/ubuntu jammy-backports/main amd64 Packages [81.0 kB]
Get:28 http://archive.ubuntu.com/ubuntu jammy-backports/universe amd64 Packages [32.2 kB]
Fetched 32.8 MB in 3s (9,987 kB/s)
Reading package lists...
W: https://download.docker.com/linux/ubuntu/dists/jammy/InRelease: Key is stored in legacy trusted.gpg keyring (/etc/apt/trusted.gpg), see the DEPRECATION section in apt-key(8) for details.
Reading package lists...
Building dependency tree...
Reading state information...
gnupg is already the newest version (2.2.27-3ubuntu2.1).
The following additional packages will be installed:
  libcurl4 libcurl4-openssl-dev
Suggested packages:
  libcurl4-doc libidn11-dev libkrb5-dev libldap2-dev librtmp-dev libssh2-1-dev
The following packages will be upgraded:
  apt-transport-https curl libcurl4 libcurl4-openssl-dev
4 upgraded, 0 newly installed, 0 to remove and 169 not upgraded.
Need to get 871 kB of archives.
After this operation, 1,024 B of additional disk space will be used.
Get:1 http://archive.ubuntu.com/ubuntu jammy-updates/universe amd64 apt-transport-https all 2.4.12 [1,510 B]
Get:2 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libcurl4-openssl-dev amd64 7.81.0-1ubuntu1.16 [386 kB]
Get:3 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 curl amd64 7.81.0-1ubuntu1.16 [194 kB]
Get:4 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 libcurl4 amd64 7.81.0-1ubuntu1.16 [290 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 871 kB in 1s (1,249 kB/s)
(Reading database ... 49215 files and directories currently installed.)
Preparing to unpack .../apt-transport-https_2.4.12_all.deb ...
Unpacking apt-transport-https (2.4.12) over (2.4.9) ...
Preparing to unpack .../libcurl4-openssl-dev_7.81.0-1ubuntu1.16_amd64.deb ...
Unpacking libcurl4-openssl-dev:amd64 (7.81.0-1ubuntu1.16) over (7.81.0-1ubuntu1.15) ...
Preparing to unpack .../curl_7.81.0-1ubuntu1.16_amd64.deb ...
Unpacking curl (7.81.0-1ubuntu1.16) over (7.81.0-1ubuntu1.15) ...
Preparing to unpack .../libcurl4_7.81.0-1ubuntu1.16_amd64.deb ...
Unpacking libcurl4:amd64 (7.81.0-1ubuntu1.16) over (7.81.0-1ubuntu1.15) ...
Setting up apt-transport-https (2.4.12) ...
Setting up libcurl4:amd64 (7.81.0-1ubuntu1.16) ...
Setting up curl (7.81.0-1ubuntu1.16) ...
Setting up libcurl4-openssl-dev:amd64 (7.81.0-1ubuntu1.16) ...
Processing triggers for libc-bin (2.35-0ubuntu3.1) ...
^@^@
* Installing APT package sources for Datadog

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1713  100  1713    0     0  17325      0 --:--:-- --:--:-- --:--:-- 17479
gpg: directory '/root/.gnupg' created
gpg: /root/.gnupg/trustdb.gpg: trustdb created
gpg: key E6266D4AC0962C7D: public key "Datadog, Inc. APT key (2023-04-20) (APT key) <package+aptkey@datadoghq.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  6226  100  6226    0     0  63242      0 --:--:-- --:--:-- --:--:-- 63530
gpg: key 32637D44F14F620E: public key "Datadog, Inc. Master key (2020-09-08) <package+masterkey@datadoghq.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  9001  100  9001    0     0   105k      0 --:--:-- --:--:-- --:--:--  105k
gpg: key D3A80E30382E94DE: public key "Datadog, Inc <package@datadoghq.com>" imported
gpg: Total number processed: 1
gpg:               imported: 1

* Installing the Datadog Agent package

Ign:1 https://apt.datadoghq.com stable InRelease
Get:2 https://apt.datadoghq.com stable Release [26.0 kB]
Get:3 https://apt.datadoghq.com stable Release.gpg [801 B]
Get:4 https://apt.datadoghq.com stable/7 all Packages [1,073 B]
Get:5 https://apt.datadoghq.com stable/7 amd64 Packages [74.9 kB]
Fetched 103 kB in 0s (750 kB/s)
Reading package lists...
  Installing package(s): datadog-agent datadog-signing-keys

Reading package lists...
Building dependency tree...
Reading state information...
The following NEW packages will be installed:
  datadog-agent datadog-signing-keys
0 upgraded, 2 newly installed, 0 to remove and 169 not upgraded.
Need to get 273 MB of archives.
After this operation, 1,261 MB of additional disk space will be used.
Get:1 https://apt.datadoghq.com stable/7 amd64 datadog-agent amd64 1:7.55.0-1 [273 MB]
Get:2 https://apt.datadoghq.com stable/7 amd64 datadog-signing-keys all 1:1.3.1-1 [22.3 kB]
debconf: delaying package configuration, since apt-utils is not installed
Fetched 273 MB in 2s (118 MB/s)
Selecting previously unselected package datadog-agent.
(Reading database ... 49215 files and directories currently installed.)
Preparing to unpack .../datadog-agent_1%3a7.55.0-1_amd64.deb ...
Unpacking datadog-agent (1:7.55.0-1) ...
Selecting previously unselected package datadog-signing-keys.
Preparing to unpack .../datadog-signing-keys_1%3a1.3.1-1_all.deb ...
Unpacking datadog-signing-keys (1:1.3.1-1) ...
Setting up datadog-agent (1:7.55.0-1) ...
Creating dd-agent group
Creating dd-agent user
Enabling service datadog-agent
Created symlink /etc/systemd/system/multi-user.target.wants/datadog-agent.service → /lib/systemd/system/datadog-agent.service.
Created symlink /etc/systemd/system/multi-user.target.wants/datadog-agent-process.service → /lib/systemd/system/datadog-agent-process.service.
Created symlink /etc/systemd/system/multi-user.target.wants/datadog-agent-trace.service → /lib/systemd/system/datadog-agent-trace.service.
Created symlink /etc/systemd/system/multi-user.target.wants/datadog-agent-security.service → /lib/systemd/system/datadog-agent-security.service.
No datadog.yaml file detected, not starting the agent
Setting up datadog-signing-keys (1:1.3.1-1) ...
W: --force-yes is deprecated, use one of the options starting with --allow instead.

* Adding your API key to the Datadog Agent configuration: /etc/datadog-agent/datadog.yaml


* Setting SITE in the Datadog Agent configuration: /etc/datadog-agent/datadog.yaml


* Adding your HOSTNAME to the Datadog Agent configuration: /etc/datadog-agent/datadog.yaml


  * DD_INSTALL_ONLY environment variable set.

    The newly installed version of the Datadog Agent will not be started.
    You will have to do it manually using the following command:
  
    sudo service datadog-agent start

datadog-agent: unrecognized service

Exited with code exit status 1
  ```
</details>

**Comment by KSerrania** [comment]
Hi @saada,

Thanks for raising this issue! I confirm that we can reproduce this error. From our initial investigation, this is affecting situations where SysVInit is being used as the init system, which happens, for instance, when installing the Agent in a container (since containers do not include newer init systems like systemd).

We are currently looking into updating the datadog-agent-orb to allow pinning the Agent version, as well as releasing a fixed version of the Agent for systems using SysVInit.

We will update this issue when we have progress to share. We will also add a note in the [release notes](https://github.com/DataDog/datadog-agent/releases/tag/7.55.0) to mention this issue.

**Comment by karlhe** [comment]
Also running into this issue, is there any workaround for this?

**Comment by jose-sherpa** [comment]
@KSerrania I am hitting this too, I thought it was an error with the install script initially but apparently its not

https://github.com/DataDog/agent-linux-install-script/issues/237

**Comment by masterplanx** [comment]
Hi, Is it possible to use something like this?

HeyIs it possible to use something like this?

```
  install-datadog:
    steps:
      - datadog-agent/setup:
          agent_major_version: '7.54.1'
          api_key: DATADOG_API_KEY
          site: datadoghq.com
```
 to specify the version that is working in the agent_major_version?

**Comment by KSerrania** [comment]
Hi @masterplanx, for users of the `datadog-agent-orb`, there is currently no workaround available. The orb uses the install script under the hood, which requires the major version to be exactly "6" or "7" (see the code [here](https://github.com/DataDog/agent-linux-install-script/blob/b74d27041e046eb6963d0ea1fb1ca396c2b831f4/install_script.sh.template#L945-L951)), and the orb does not provide another way to specify the minor version to use. For these cases, we're working on releasing an updated version of the orb.

Hi @karlhe and @jose-sherpa, if you are using the install script directly, as a workaround I recommend pinning the previous working version of the Agent, `7.54.1`, by adding the `DD_AGENT_MINOR_VERSION=54.1` environment variable to your install script command. Example command:

```
DD_HOSTNAME=test DD_API_KEY=xxxx DD_AGENT_MINOR_VERSION=54.1 DD_SITE="datadoghq.com"  bash -c "$(curl -L https://install.datadoghq.com/scripts/install_script_agent7.sh)
```

**Comment by jose-sherpa** [comment]
Yep @KSerrania thats what we are doing for now, thank you!

**Comment by KSerrania** [comment]
Hi @masterplanx, we just published [a new version](https://circleci.com/developer/orbs/orb/datadog/agent) of the `datadog-agent-orb`, `3.2.1`. It contains two updates:
- it allows specifying a minor version, with the `agent_minor_version` parameter.
- it defaults the `agent_minor_version` parameter to `54.1`. This way, the orb uses a known good version of the Agent by default.

Please bump the version of the orb to `3.2.1` to benefit from this workaround, by specifying:
```
orbs:
  datadog-agent: datadog/agent@3.2.1
```
in your CircleCI configuration.

**Comment by masterplanx** [comment]
Great news! thank you I will try

**Comment by masterplanx** [comment]
@KSerrania is wroking well ! thank you

**Comment by simonh-tecex** [comment]
Please do fix, this is impacting our build processes as well. We are able to work-around the issue for the moment.

**Comment by spencergilbert** [comment]
Hi all,

We've just released [a new version](https://github.com/DataDog/datadog-agent/releases/tag/7.55.1) of the `datadog-agent`, `7.55.1`. This release fixes a regression where the agent would fail to start on systems with SysVinit.

Using `7.55.1` should remove the need for any workaround provided above. If using the `datadog-agent-orb` at `3.2.1` and you want to use the latest Agent version, you'll need set the `agent_minor_version` parameter to `55.1`. An upcoming release of the orb will again target the latest Agent release by default.

**Comment by saada** [comment]
Confirming the released fix in 3.2.1 resolves the issue. Thanks team
